### PR TITLE
[mjml-react] Add Explicit types for children

### PR DESCRIPTION
**Pull request by eps1lon**
We plan to remove implicit children from `@types/react`. The following changes are required to pass https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56210.

https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-wrapper.js#L8
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-breakpoint.js#L10
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-font.js#L7
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-button.js#L8
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-column.js#L8
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-divider.js#L7
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-group.js#L8
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-hero.js#L8
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-image.js#L7
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-navbar.js#L8
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-navbar-link.js#L8
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-spacer.js#L7
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-table.js#L8
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-wrapper.js#L8
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-carousel.js#L8
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-carousel-image.js#L10
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-social-element.js#L8
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-social.js#L8
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-section.js#L8
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-column.js#L8
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-button.js#L8
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-text.js#L8
https://github.com/wix-incubator/mjml-react/blob/c1eb06ab366697a609dc6f552f86855b03ae4d80/src/mjml-body.js#L8

### CODE REVIEW

**Comment by eps1lon** [comment]
> IMO there's isn't enough detail about this change to evaluate what the goal is.

What details are you missing? I've provided a link to the overall goal and describe each individual component. I don't know what else I could provide.

> This also changes tests in undesirable ways.

Could you point to specifics? I'm not familiar with what you desire in your tests.

**Review by IanEdington — CHANGES_REQUESTED** [review]

File: types/mjml-react/mjml-react-tests.tsx [hunk 1]
```diff
@@ -115,11 +115,11 @@ function renderOutTestEmail() {
 // TestMjmlBreakpointTag
 {
     const minProps: React.ReactNode = <MjmlBreakpoint />;
-    const maxProps: React.ReactNode = <MjmlBreakpoint width="">child</MjmlBreakpoint>;
+    const maxProps: React.ReactNode = <MjmlBreakpoint width="" />;
 }
 // TestMjmlBodyTag
 {
-    const minProps: React.ReactNode = <MjmlBody />;
+    const minProps: React.ReactNode = <MjmlBody />; // $ExpectError
```
[the remark this answers is not in the document]
Comment (IanEdington) [reply types/mjml-react/mjml-react-tests.tsx:122]: @eps1lon the PR upstream has merged. Are you able to loosen the types in this PR.

My expectation is that you shouldn't need to change any tests from the base branch for this change. Does that make sense or is there something I'm still missing about the test changes?

[the remark this answers is not in the document]
Comment (eps1lon) [reply types/mjml-react/mjml-react-tests.tsx]: Updated types with https://github.com/wix-incubator/mjml-react/pull/66

**Review by IanEdington — APPROVED** [review]

**Comment by eps1lon** [comment]
Ready to merge

**Pull request merged**